include/RAJA/policy/cuda/raja_cudaerrchk.hpp

```diff
@@ -43,33 +109,95 @@ namespace RAJA
+    storage_type{"cudaGetDevice",                                 "device"},
+    storage_type{"cudaGetDeviceProperties",                       "prop device"},
+    storage_type{"cudaStreamSynchronize",                         "stream"},
+    storage_type{"cudaHostAlloc",                                 "pHost size flags"},
+    storage_type{"cudaHostFree",                                  "ptr"},
+    storage_type{"cudaMalloc",                                    "devPtr size"},
+    storage_type{"cudaFree",                                      "devPtr"},
+    storage_type{"cudaMemset",                                    "devPtr value count"},
+    storage_type{"cudaMemcpy",                                    "dst src count kind"},
+    storage_type{"cudaMemsetAsync",                               "devPtr value count stream"},
+    storage_type{"cudaMemcpyAsync",                               "dst src count kind stream"},
+    storage_type{"cudaLaunchKernel",                              "func gridDim blockDim args sharedMem stream"},
+    storage_type{"cudaPeekAtLastError",                           ""},
+    storage_type{"cudaGetLastError",                              ""},
+    storage_type{"cudaFuncGetAttributes",                         "attr func"},
+    storage_type{"cudaOccupancyMaxPotentialBlockSize",            "minGridSize blockSize func dynamicSMemSize blockSizeLimit"},
+    storage_type{"cudaOccupancyMaxActiveBlocksPerMultiprocessor", "numBlocks func blockSize dynamicSMemSize"}
+  }};
+  for (auto [api_name, api_args] : known_functions) {
+    if (func == api_name) {
+      return api_args;
+    }
+  }
+  return "";
+}
+
+template < typename Tuple >
+RAJA_INLINE void cudaAssert(cudaError_t code,
+                            std::string_view func_name,
+                            std::string_view args_name,
+                            Tuple const& args,
+                            std::string_view file,
+                            int line,
+                            bool abort = true)
 {
   if (code != cudaSuccess)
   {
+    std::ostringstream str;
+    str << "CUDAassert: ";
+    str << cudaGetErrorString(code);
+    str << " ";
+    str << func_name;
+    str << "(";
+    const auto args_end = args_name.end();
+    for_each_tuple(args, [&, first=true, args_current=args_name.begin()](auto&& arg) mutable {
+      if (!first) {
+        str << ", ";
+      } else {
+        first = false;
+      }
+      if (args_current != args_end) {
+        auto args_current_end = std::find(args_current, args_end, ' ');
+        str << std::string_view{args_current, size_t(args_current_end-args_current)} << "=";
+        if (args_current_end != args_end) {
+          ++args_current_end; // skip space
+        }
+        args_current = args_current_end;
+      }
+      str << ::RAJA::detail::StreamInsertHelper{arg};
+    });
+    str << ") ";
+    str << file;
+    str << ":";
+    str << line;
+    auto msg{str.str()};
```

Comment:
Do you have an example of what a full error message looks like now?

Reply:
```
unknown file: Failure
C++ exception with description "HIPassert: invalid configuration argument hipLaunchKernel(func=0x273ff0, gridDim={3,2,1000000}, blockDim={3,2,1000000}, args=0x7fffffff76f8, sharedMem=0, stream=0x7e0860) /g/g19/burmark1/libraries/RAJA/include/RAJA/policy/hip/MemUtils_HIP.hpp:273" thrown in the test body.
```